src/core/lib/event_engine/posix_engine/posix_engine_listener.h

```diff
@@ -121,6 +126,8 @@ class PosixEngineListenerImpl
     ListenerSocketsContainer::ListenerSocket socket_;
     EventHandle* handle_;
     PosixEngineClosure* notify_on_accept_;
+    PosixEngineClosure* retry_closure_;
```

Comment:
retry_closure also needs to be deleted explicitly since its a permanent closure. Otherwise it'll be a memory leak. Maybe it can be a self deleting closure instead of a permanent closure ?  Or maybe the timer needs to be cancelled and closure needs to be deleted in the destructor

Reply:
As discussed, I replaced it with an AnyInvocable.